src/toDoList.js

```diff
@@ -2,39 +2,153 @@ import React from 'react';
 
 class ToDoList extends React.Component {
   constructor() {
-      super();
-      this.state = {
+      super(); //it calls react Component constructor.
+      this.state = { //state is a plain js object..init only in constructor
         items: [],
-        value: ''
+        value: '',
+        isUpdate: false,
+        id: 9999,
+        searchValue : '',
+        searchResults: []
       };
       this.addItem = this.addItem.bind(this);
       this.displayList = this.displayList.bind(this);
+      this.removeItem = this.removeItem.bind(this);
+      this.onChangeFunctions = this.onChangeFunctions.bind(this);
+      this.doRenderSearchBar =  this.doRenderSearchBar.bind(this);
+  }
+
+  editItem(id) {
+    this.setState({
+      isUpdate : true,
+      id : id
+    });
   }
 
   addItem() {
-    let inputText = this.state.value;
-    this.state.items = [...this.state.items, inputText];
-    this.state.value = '';
+    let inputText = this.state.value; //object destructring
+    let newItems = this.state.items;
+    let searchItems = this.state.searchResults;
+    let isCheck = false;
+
+    if(this.state.isUpdate) {
+      newItems = newItems.map((value, i) => {
+        if(value === searchItems[this.state.id] && !isCheck) {
+          isCheck = true;
+          return inputText;
+        }
+        return value;
+      });
+      searchItems[this.state.id] = inputText;
+    } else {
+      newItems = [...this.state.items, inputText]
+      searchItems = [...newItems]
+    }
+    //this.state.value = '';  dont change state variables directly... use setState instead
     this.setState({
       value : '',
-      items : this.state.items
+      items : newItems,
+      isUpdate : false,
+      id : 9999,
+      searchResults: searchItems
     });
   }
 
-  displayList(){
-    return (this.state.items.slice(0).reverse().map((item,i) => <li key={i}>{item}</li>));
+  removeItem(id) {
+    let isCheck = false;
+    let newItems = this.state.items.filter((item, i) => {
+      if(item === this.state.searchResults[id] && !isCheck) {
+        isCheck = true;
```

Comment:
If you are using filter then, it should return something, and here it isn't.

Reply:
wanted to return  a single result from .filter, hence used isCheck flag in combination. Alternative way to do this ?

Reply:
Use For Loop.